src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -1047,6 +977,34 @@ def perform_checks(self, analyze_ctx: AnalyzeContext) -> dict[str, CheckResult]:
 
         return results
 
+    def _resolve_more_dependencies(self, dependencies: dict[str, DependencyInfo]) -> None:
+        """Utilise the Repo Finder to resolve the repositories of more dependencies."""
+        for item in dependencies.values():
+            if item.get("available") != SCMStatus.MISSING_SCM:
+                continue
+
+            name = item.get("name") or ""
```

Comment:
This part is probably related to [this discussion](https://github.com/oracle/macaron/pull/388/files/57d13c9d300bbe896b8c315670ef5accfc3e2333#r1312331740): if we store PURL in `DependencyInfo`, we wouldn't need to reconstruct PURLs here.